src/components/SortBy.jsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from "react"
+import { useSearchParams } from "react-router-dom"
+
+const SortBy = (props) => {
+
+    const [searchParams, setSearchParams] = useSearchParams(props)
+
+    // pass via props
+    const [topic, setTopic] = useState()
+    const [sortByQuery, setSortByQuery] = useState()
+
+    const setNewTopic = (event) => {
+        props.setTopic(event.target.value)
+    }
+
+    const setNewParams = (event) => {
+        let queryOrder = event.target.value.split("__")
+            props.setSortBy(queryOrder[0])
+            props.setOrder(queryOrder[1])
+
+    }
+
+    return (
+    <div className="sort-by-bar">
+        <form >
+                <label htmlFor="topic-selector">sort by...</label>
+                <select id="topic-selector" name="topic-selector" value={topic} onChange={setNewTopic}>
+                    <option value={""}>All Topics</option>
+                    <option value={"coding"}>Coding</option>
+                    <option value={"cooking"}>Cooking</option>
+                    <option value={"football"}>Football</option>
+
+                </select>
+            </form>
+
+        <form >
+                <label htmlFor="sort-by">sort by...</label>
+                <select id="sort=by" name="sort-by" value={sortByQuery} onChange={setNewParams}>
+                    <option value={"articles.created_at__DESC"}>Date (newest first)</option>
+                    <option value={"articles.created_at__ASC"}>Date (oldest first)</option>
+                    <option value={"comment_count__DESC"}>Most Comments</option>
+                    <option value={"comment_count__ASC"}>Least Comments</option>
+                    <option value={"articles.votes__DESC"}>Most Popular</option>
+                    <option value={"articles.votes__ASC"}>Least Popular</option>
```

Comment:
Could consider having a button to control the `order` (like a toggle) and a dropdown to control the `sort_by`
(this is very much optional, just a suggestion, but may be a bit more preferable for UX)